[2025-07-04T20:01:32.132+0000] {taskinstance.py:1087} INFO - Dependencies all met for <TaskInstance: retail_sales_pipeline.simulate_kafka_events manual__2025-07-04T19:41:01.587132+00:00 [queued]>
[2025-07-04T20:01:32.148+0000] {taskinstance.py:1087} INFO - Dependencies all met for <TaskInstance: retail_sales_pipeline.simulate_kafka_events manual__2025-07-04T19:41:01.587132+00:00 [queued]>
[2025-07-04T20:01:32.148+0000] {taskinstance.py:1283} INFO - 
--------------------------------------------------------------------------------
[2025-07-04T20:01:32.149+0000] {taskinstance.py:1284} INFO - Starting attempt 3 of 4
[2025-07-04T20:01:32.149+0000] {taskinstance.py:1285} INFO - 
--------------------------------------------------------------------------------
[2025-07-04T20:01:32.176+0000] {taskinstance.py:1304} INFO - Executing <Task(BashOperator): simulate_kafka_events> on 2025-07-04 19:41:01.587132+00:00
[2025-07-04T20:01:32.188+0000] {standard_task_runner.py:82} INFO - Running: ['***', 'tasks', 'run', 'retail_sales_pipeline', 'simulate_kafka_events', 'manual__2025-07-04T19:41:01.587132+00:00', '--job-id', '13', '--raw', '--subdir', 'DAGS_FOLDER/retail_sales_pipeline.py', '--cfg-path', '/tmp/tmpykz1wizi']
[2025-07-04T20:01:32.187+0000] {standard_task_runner.py:55} INFO - Started process 946 to run task
[2025-07-04T20:01:32.190+0000] {standard_task_runner.py:83} INFO - Job 13: Subtask simulate_kafka_events
[2025-07-04T20:01:32.519+0000] {task_command.py:389} INFO - Running <TaskInstance: retail_sales_pipeline.simulate_kafka_events manual__2025-07-04T19:41:01.587132+00:00 [running]> on host ce609c2457b1
[2025-07-04T20:01:32.832+0000] {taskinstance.py:1513} INFO - Exporting the following env vars:
AIRFLOW_CTX_DAG_OWNER=***
AIRFLOW_CTX_DAG_ID=retail_sales_pipeline
AIRFLOW_CTX_TASK_ID=simulate_kafka_events
AIRFLOW_CTX_EXECUTION_DATE=2025-07-04T19:41:01.587132+00:00
AIRFLOW_CTX_TRY_NUMBER=3
AIRFLOW_CTX_DAG_RUN_ID=manual__2025-07-04T19:41:01.587132+00:00
[2025-07-04T20:01:32.834+0000] {subprocess.py:63} INFO - Tmp dir root location: 
 /tmp
[2025-07-04T20:01:32.835+0000] {subprocess.py:75} INFO - Running command: ['/bin/bash', '-c', '\n        python3 /opt/***/scripts/ingestion/kafka_producer.py\n        ']
[2025-07-04T20:01:32.845+0000] {subprocess.py:86} INFO - Output:
[2025-07-04T20:01:33.219+0000] {subprocess.py:93} INFO - Traceback (most recent call last):
[2025-07-04T20:01:33.221+0000] {subprocess.py:93} INFO -   File "/opt/***/scripts/ingestion/kafka_producer.py", line 35, in <module>
[2025-07-04T20:01:33.222+0000] {subprocess.py:93} INFO -     sale_event = generate_sale_event()
[2025-07-04T20:01:33.222+0000] {subprocess.py:93} INFO -   File "/opt/***/scripts/ingestion/kafka_producer.py", line 18, in generate_sale_event
[2025-07-04T20:01:33.223+0000] {subprocess.py:93} INFO -     "customer_id": random.choice(1,73),
[2025-07-04T20:01:33.223+0000] {subprocess.py:93} INFO - TypeError: choice() takes 2 positional arguments but 3 were given
[2025-07-04T20:01:33.249+0000] {subprocess.py:97} INFO - Command exited with return code 1
[2025-07-04T20:01:33.322+0000] {taskinstance.py:1772} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/operators/bash.py", line 197, in execute
    f"Bash command failed. The command returned a non-zero exit code {result.exit_code}."
airflow.exceptions.AirflowException: Bash command failed. The command returned a non-zero exit code 1.
[2025-07-04T20:01:33.329+0000] {taskinstance.py:1327} INFO - Marking task as UP_FOR_RETRY. dag_id=retail_sales_pipeline, task_id=simulate_kafka_events, execution_date=20250704T194101, start_date=20250704T200132, end_date=20250704T200133
[2025-07-04T20:01:33.368+0000] {standard_task_runner.py:105} ERROR - Failed to execute job 13 for task simulate_kafka_events (Bash command failed. The command returned a non-zero exit code 1.; 946)
[2025-07-04T20:01:33.410+0000] {local_task_job.py:159} INFO - Task exited with return code 1
[2025-07-04T20:01:33.514+0000] {taskinstance.py:2582} INFO - 0 downstream tasks scheduled from follow-on schedule check
